crates/miden-tx/src/executor/exec_host.rs

```diff
@@ -277,24 +277,21 @@ where
 
     /// Handles a request for a storage map witness by querying the data store for a merkle path.
     ///
-    /// Note that we request witnesses against the initial map root for native accounts. See also
+    /// Note that we request witnesses against the _initial_ map root of the accounts. See also
     /// [`Self::on_account_vault_asset_witness_requested`] for more on this topic.
     async fn on_account_storage_map_witness_requested(
         &self,
         current_account_id: AccountId,
         slot_index: usize,
-        _map_root: Word,
+        current_map_root: Word,
```

Comment:
Is passing both `slot_index` and `current_map_root` a bit redundant here? Meaning, on any given invocation, we care either about `slot_index` (in case of the native account) or `current_map_root` (in case of the foreign account).

Would it be possible to only pass the `map_root` here? This would require figuring out if we are dealing with the native account a bit earlier - and I haven't looked deeply enough to know if that causes any significant complications.